### PR TITLE
Further optimization of ES force calculation.

### PR DESCRIPTION
**Pull request by nbrunk**
In the ES force loop, I removed use of the Grad() function, as it requires separate calculation of "r_vec" and some other things, in favor of using those already computed.

A histogram of how the different steps affect run speed (in steps per second normalized by the base code we have been using).  Different first numbers indicate things done separately (aside from 5, which is a Collective estimate when all things done in the previous pull request are done simultaneously).

Capture

So with just the both sets of ES force loop optimizations (Steps 4.1 and 6), there's about a 40% speed-up.  If this scales the same, each job will take around 3.5h or less on the cluster.

### CODE REVIEW

**Comment by nbrunk** [comment]
This has yields the same results as previously for the disc control:

Control_Matching

**Comment by nbrunk** [comment]
@jadhao Please approve.  40% speedup

File: src/functions.h [hunk 1]
```diff
@@ -221,6 +204,21 @@ void initialize_vertex_velocities_to_zero(vector<VERTEX> &V);
 
 // ### Unused functions: ###
```
Comment (jadhao) [inline src/functions.h]: @nbrunk are these really unused; they are not used by bending?

**Comment by nbrunk** [comment]
@jadhao No, they are not used.  CLion reports if functions are used or not.

**Comment by nbrunk** [comment]
@jadhao Please approve.

File: src/newforces.cpp [hunk 2]
```diff
@@ -85,9 +88,13 @@ void force_calculation(INTERFACE &boundary, const double scalefactor) {
             VECTOR3D ljforce;
             VECTOR3D esforce;
             VECTOR3D r_vec = boundary.V[i].posvec - boundary.V[j].posvec;
+            double r = r_vec.GetMagnitude();
             double r2 = r_vec.GetMagnitudeSquared();
+            double r3 = r * r * r;
```
Comment (jadhao) [inline src/newforces.cpp]: @nbrunk I think you do not need r3. only r and r2 appear when you factorize (see below).

Comment (nbrunk) [reply src/newforces.cpp]: @jadhao Line 28 for defining "grad_vec" has an "r3" in it.  Maybe I'm missing something?

File: src/functions.h (hunk 1 above)
Comment (nbrunk) [reply src/functions.h]: @jadhao No, they are not used. CLion reports if functions are used anywhere in the code or not.  Moving my comment into this conversation thread, then resolving.

File: src/newforces.cpp (hunk 2 above)
Comment (nbrunk) [reply src/newforces.cpp]: @jadhao This has been added, circumventing explicit definition of the "grad_vec".

**Comment by nbrunk** [comment]
@jadhao I've factored such that we avoid computing "r3", circumventing explicit definition of "grad_vec" as well.  This has been checked to give the same results:

Control_Matching

@kadupitiya You can double check the above if you like, I always use the standard disc control.  See the README.

**Comment by jadhao** [comment]
@nbrunk approving
@kadupitiya please provide a double check

**Comment by kadupitiya** [comment]
@nbrunk, @jadhao: I checked examples vs bigred2. Energies are intact. Up to 25,000, it is 100% matching.

**Pull request merged**